src/fromager/bootstrapper.py

```diff
@@ -432,40 +499,54 @@ def _download_wheel_from_cache(
                 server.update_wheel_mirror(self.ctx)
             logger.info(f"{req.name}: found built wheel on cache server")
-            unpack_dir = self._create_unpack_dir(req, resolved_version)
-            dist_filename = f"{dist_name}-{dist_version}"
-            metadata_dir = pathlib.Path(f"{dist_filename}.dist-info")
-            try:
-                archive = zipfile.ZipFile(cached_wheel)
-                for filename in [
-                    dependencies.BUILD_BACKEND_REQ_FILE_NAME,
-                    dependencies.BUILD_SDIST_REQ_FILE_NAME,
-                    dependencies.BUILD_SYSTEM_REQ_FILE_NAME,
-                ]:
-                    zipinfo = archive.getinfo(
-                        str(
-                            metadata_dir
-                            / f"{wheels.FROMAGER_BUILD_REQ_PREFIX}-{filename}"
-                        )
-                    )
-                    zipinfo.filename = filename
-                    archive.extract(zipinfo, unpack_dir)
-
-                logger.info(f"{req.name}: extracted build requirements from wheel")
-                return unpack_dir / metadata_dir, cached_wheel
-            except Exception:
-                # implies that the wheel server hosted non-fromager built wheels
-                logger.info(
-                    f"{req.name}: could not extract build requirements from wheel"
-                )
-                shutil.rmtree(unpack_dir)
-                return None, cached_wheel
+            unpack_dir = self._unpack_metadata_from_wheel(
+                req, resolved_version, cached_wheel
+            )
+            return cached_wheel, unpack_dir
         except Exception:
             logger.info(
                 f"{req.name}: did not find wheel for {resolved_version} in {self.cache_wheel_server_url}"
             )
             return None, None
 
+    def _unpack_metadata_from_wheel(
+        self, req: Requirement, resolved_version: Version, wheel_filename: pathlib.Path
+    ) -> pathlib.Path | None:
+        dist_name, dist_version, build_tag, _ = wheels.extract_info_from_wheel_file(
+            req,
+            wheel_filename,
+        )
+        unpack_dir = self._create_unpack_dir(req, resolved_version)
+        dist_filename = f"{dist_name}-{dist_version}"
+        metadata_dir = pathlib.Path(f"{dist_filename}.dist-info")
+        output_dir = unpack_dir / metadata_dir
+        output_dir.mkdir(parents=True, exist_ok=True)
+        try:
+            archive = zipfile.ZipFile(wheel_filename)
+            for filename in [
+                dependencies.BUILD_BACKEND_REQ_FILE_NAME,
+                dependencies.BUILD_SDIST_REQ_FILE_NAME,
+                dependencies.BUILD_SYSTEM_REQ_FILE_NAME,
+            ]:
+                zipinfo = archive.getinfo(
+                    str(metadata_dir / f"{wheels.FROMAGER_BUILD_REQ_PREFIX}-{filename}")
+                )
+                zipinfo.filename = filename
```

Comment:
Why is the code assigning to `zipinfo.filename` here?

Reply:
It was doing that before, I just moved the code around. I'm not sure why it would have needed to do that.

Reply:
I think I see what it's doing. Inside the zip file the metadata files have a prefix to ensure they are unique and marked as coming from fromager. When they are unpacked, that prefix is being removed so the files match the names used when fromager generates similar files during a bootstrap run and to unpack them without the metadata dir in the path. 

We could change that, but I'd like to do it in a separate PR because it will be a bit invasive.